### ISSUE TITLE
Chrome browser on Android

### ISSUE THREAD

**Issue by obiot** [goldfire/howler.js#518]
Using the latest Howler 2.0 beta 12 through melonJS we had an issue reported when using Chrome on Android : mostly when forcing/setting the `html5` flag to true when creating a new Howl object (loading then stall)  

I'm sorry I do not really have a test case, or better description, but does anyone has experience with it ?

**Comment by goldfire** [comment]
Heard anymore on this? I've tinkered around, but without much to go on I haven't been able to find an issue.

**Comment by agentfitz** [comment]
Hey guys, I figured out what was going on with this (at least in my code).

Writing module js with browserify, I had one module (`player-controls.js`) which was listening for a click on the play button and firing a custom event `player_controls_play_clicked`.

In turn, I had my `audio-playback.js` module (wrapping Howler) listening for the `player_controls_play_clicked` event and then calling `sound.play()`. This causes a js error on Android as a console error indicates that [play can only be initiated by a user gesture](http://stackoverflow.com/questions/32424775/failed-to-execute-play-on-htmlmediaelement-api-can-only-be-initiated-by-a-u).

It is less than ideal from a coupling standpoint, but adding the play button click handler directly into my `audio-playback.js` module and having it call `sound.play()` without having to respond to the intermediary custom event, solved the issue for me. Haven't tested on iOS.

So to recap, to ensure reliable playback on mobile devices, make sure that the code calling `sound.play()` is directly responding to user actions without relying on any intermediary code.

**Comment by LaPieza** [comment]
Hello, I'm the one who experienced this issue on MelonJS.

I use MelonJS to load the audio, and it works good except when I try to put html : true  to allow streaming, everything works ok except on some browsers on Android, for example it works on Firefox on Android but not on Chrome.

When I put the html : true that sound never plays, but the others do, I haven't tried Agentfitz solution because I need the audio to self start because I'm making a game for Android TV soo It wont look good if I put a button that needs to be pressed on the screen (didn't worked to make it play after phisicial button pressed).

Also I use the html : true to make the sounds load faster (the sounds will be stored localy as I will make an apk), the files that I try to load this way are the background music of the game soo waiting for them to load without the html : true is takes too much time.

**Comment by goldfire** [comment]
I'm not sure if it works differently on on Android TV (one would hope so), but on Chrome for Android any HTML5 sound must be initiated within a user action. I'm not currently aware of any way to get around this.

**Comment by LaPieza** [comment]
Hello Goldfire, sound does auto start using howler without the need to touch the screen on Chrome for Android, but only when not using -> html : true

The downside of not using html : true, is that it needs more time to load the sounds, which is not too much for efects but for various tracks of background it does takes much time.

Also on Chromium (building an apk with Crosswalk or CocoonIO) the audio does autostart when using a regular hmtl audio tag (that's what I'm doing right now for long sounds).

**Comment by goldfire** [comment]
Right, the behavior is different between HTML5 Audio and Web Audio in Chrome on Android.

**Comment by LaPieza** [comment]
Oh, I see.

**Comment by goldfire** [comment]
I am sure I had come across an issue in the Chromium tracker about this, though I can't find it again at the moment.

**Comment by tomasz-swirski** [comment]
https://github.com/goldfire/howler.js/pull/336

**Comment by mvinayakam** [comment]
A sideeffect of this issue is that html5 audio pauses when an incoming call is detected whereas the web Audio continues playing through the call

**Comment by just1witness** [comment]
> Hey guys, I figured out what was going on with this (at least in my code).

Writing module js with browserify, I had one module (player-controls.js) which was listening for a click on the play button and firing a custom event player_controls_play_clicked.

In turn, I had my audio-playback.js module (wrapping Howler) listening for the player_controls_play_clicked event and then calling sound.play(). This causes a js error on Android as the console reports that play can only be initiated by a user gesture.

It is less than ideal from a coupling standpoint, but adding the play button click handler directly into my audio-playback.js module and having it call sound.play() without having to respond to the intermediary custom event, solved the issue for me. Haven't tested on iOS.

So to recap, to ensure reliable playback on mobile devices, make sure that the code calling sound.play() is directly responding to user actions without relying on any intermediary code.

**So that means, howler can't be used as an alarm for a timer? Since there has to be intermediary code to determine if the timer reached 0? (assuming android chrome compatibility is a must)**

**Comment by goldfire** [comment]
Closing due to inactivity. If this is still an issue, reply with more info and I'll reopen.
